lib/credentials/shared_ini_file_credentials.js

```diff
@@ -87,6 +95,70 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   /**
    * @api private
    */
+  loadRoleProfile: function loadRoleProfile(creds, callback) {
+    var self = this;
+    var roleArn = creds[this.profile]['role_arn'];
+    var roleSessionName = creds[this.profile]['role_session_name'];
+    var externalId = creds[this.profile]['external_id'];
+    var sourceProfileName = creds[this.profile]['source_profile'];
+
+    if (!sourceProfileName) {
+      throw new Error('source_profile is not set in ' + this.filename +
+                      ' using profile ' + this.profile);
+    }
+
+    var sourceProfile = creds[sourceProfileName];
+
+    if (typeof creds[this.profile] !== 'object') {
```

Comment:
This was a bug. The condition should have been checking `sourceProfile`. It has been fixed.